### ISSUE TITLE
aria-label incorrectly set on markers with no role attribute causing accessibility issues

### ISSUE THREAD

**Issue by cmburcus** [maplibre/maplibre-gl-js#6435]
Map markers use `div` elements with an `aria-label="Map marker"` and no role which is not recommended from an accessibility standpoint. Markers should either not use `div`, a proper `role` attribute should be assigned or no `aria-label` should be set.

**maplibre-gl-js version**: 5.7.2

**browser**: All

### Steps to Trigger Behavior

 1. Render a marker on the map
 2. Inspect marker

### Link to Demonstration

https://maplibre.org/maplibre-gl-js/docs/examples/add-a-default-marker/

The issue can be clearly seen on the maplibre example by just inspecting the marker on the map. 

<img width="365" height="78" alt="Image" src="https://github.com/user-attachments/assets/54aad361-d8d9-4e7c-809d-509e4da436f5" />

### Expected Behavior

`div` with `role` attribute and `aria-label` OR element that accepts `aria-label` OR no `aria-label` supplied

### Actual Behavior

`div` with no `role` and default `aria-label` of "Map marker"

**Comment by HarelM** [comment]
Feel free to open a PR to fix the issue, the marker code is contained within the marker.ts class. 
Since I'm not an expert in accessibility it would help a lot of you could help here.

**Comment by cmburcus** [comment]
Thank you @HarelM . I shall take a look at the contribution readme and try to follow along!

**Comment by cmburcus** [comment]
@HarelM i'm trying to push a branch with the initial test as per the contribution readme file. However, I'm getting permission denied. Would you be able to direct me on what i need to do? I've set the upstream correctly.

```
ERROR: Permission to maplibre/maplibre-gl-js.git denied to cmburcus.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by HarelM** [comment]
I would advise to create a fork and update there if you haven't done so already...?

**Comment by cmburcus** [comment]
> I would advise to create a fork and update there if you haven't done so already...?

That was silly of me. It's very clearly stated in the readme 😅 Sorry, this is my first contribution to an open source repo.
